test/suites/dev/test-xcm-v2/test-mock-hrmp-transact-ethereum-2.ts

```diff
@@ -0,0 +1,151 @@
+      title: "should receive transact and should be able to execute",
+      test: async function () {
+        // Get Pallet balances index
+        const metadata = await context.polkadotJs().rpc.state.getMetadata();
+        const balancesPalletIndex = metadata.asLatest.pallets
+          .find(({ name }) => name.toString() == "Balances")!
+          .index.toNumber();
+
+        const xcmTransactions = [
+          {
+            V1: {
+              gas_limit: 100000,
+              fee_payment: {
+                Auto: {
+                  Low: null,
+                },
+              },
+              action: {
+                Call: contractDeployed,
+              },
+              value: 0n,
+              input: encodeFunctionData({
+                abi: contractABI,
+                functionName: "incr",
+                args: [],
+              }),
+              access_list: null,
+            },
+          },
+          {
+            V2: {
+              gas_limit: 100000,
+              action: {
+                Call: contractDeployed,
+              },
+              value: 0n,
+              input: encodeFunctionData({
+                abi: contractABI,
+                functionName: "incr",
+                args: [],
+              }),
+              access_list: null,
+            },
+          },
+        ];
+
+        let expectedCalls = 0n;
+
+        for (const xcmTransaction of xcmTransactions) {
+          expectedCalls++;
+
+          // TODO need to update lookup types for xcm ethereum transaction V2
+          const transferCall = context.polkadotJs().tx.ethereumXcm.transact(xcmTransaction as any);
+          const transferCallEncoded = transferCall?.method.toHex();
+          // We are going to test that we can receive a transact operation from parachain 1
+          // using descendOrigin first
+          const xcmMessage = new XcmFragment({
+            assets: [
+              {
+                multilocation: {
+                  parents: 0,
+                  interior: {
+                    X1: { PalletInstance: balancesPalletIndex },
+                  },
+                },
+                fungible: transferredBalance / 2n,
+              },
+            ],
+            weight_limit: new BN(4000000000),
+            descend_origin: sendingAddress,
+          })
+            .descend_origin()
+            .withdraw_asset()
+            .buy_execution()
+            .push_any({
+              Transact: {
+                originType: "SovereignAccount",
+                requireWeightAtMost: new BN(3000000000),
+                call: {
+                  encoded: transferCallEncoded,
+                },
+              },
+            })
+            .as_v2();
+
+          // Send an XCM and create block to execute it
+          await injectHrmpMessageAndSeal(context, 1, {
+            type: "XcmVersionedXcm",
+            payload: xcmMessage,
+          } as RawXcmMessage);
+
+          const actualCalls = (
```

Comment:
We have a much nicer way of doing this. Using the following:
```
await context.readContract!({
              contractAddress: contractDeployed,
              contractName: "Incrementor",
              functionName: "count",
            })
```

Will return the decoded value without having to deal with boilerplate.